Generator/Sources/NeedleFramework/Generating/Pluginized/PluginizedDependencyProviderSerializerTask.swift

```diff
@@ -59,31 +59,51 @@ class PluginizedDependencyProviderSerializerTask: AbstractTask<[SerializedProvid
+            let attributes = calculateAttributes(for: provider.data, funcNameSerializer: funcNameSerializer)
+            result.append(SerializedProvider(content: content, registration: registration, attributes: attributes))
         }
         return result
     }
 
-    private func serializedContent(for provider: PluginizedProcessedDependencyProvider, baseClassSerializer: Serializer) -> String {
-        let classNameSerializer = DependencyProviderClassNameSerializer(provider: provider.data)
-        let initBodySerializer = DependencyProviderInitBodySerializer(provider: provider.data)
-
-        let serializer = DependencyProviderSerializer(provider: provider.data, classNameSerializer: classNameSerializer, baseClassSerializer: baseClassSerializer, initBodySerializer: initBodySerializer)
-        return serializer.serialize()
+    private func serializedContent(for provider: PluginizedProcessedDependencyProvider, classNameSerializer: Serializer, paramsSerializer: Serializer, funcNameSerializer: Serializer) -> String {
+        if provider.data.isEmptyDependency {
+            return ""
+        }
+        return DependencyProviderFuncSerializer(provider: provider.data, funcNameSerializer: funcNameSerializer, classNameSerializer: classNameSerializer, paramsSerializer: paramsSerializer).serialize()
     }
 
     private func serializedBase(for provider: PluginizedProcessedDependencyProvider, counter: Int) -> (Serializer, String) {
-        let classNameSerializer = DependencyProviderBaseClassNameSerializer(provider: provider.data)
+        let classNameSerializer = DependencyProviderClassNameSerializer(provider: provider.data)
         let propertiesSerializer = PluginizedPropertiesSerializer(provider: provider)
         let sourceComponentsSerializer = SourceComponentsSerializer(componentTypes: provider.data.levelMap.keys.sorted())
         let initBodySerializer = DependencyProviderBaseInitSerializer(provider: provider.data)
 
-        let serializer = DependencyProviderBaseSerializer(provider: provider.data, classNameSerializer: classNameSerializer, propertiesSerializer: propertiesSerializer, sourceComponentsSerializer: sourceComponentsSerializer, initBodySerializer: initBodySerializer)
+        let serializer = DependencyProviderClassSerializer(provider: provider.data, classNameSerializer: classNameSerializer, propertiesSerializer: propertiesSerializer, sourceComponentsSerializer: sourceComponentsSerializer, initBodySerializer: initBodySerializer)
         return (classNameSerializer, serializer.serialize())
     }
+
+    private func calculateAttributes(for provider: ProcessedDependencyProvider, funcNameSerializer: Serializer) -> [String: String] {
+        if provider.isEmptyDependency {
+            return [:]
+        }
+        var maxLevel: Int = 0
+        provider.levelMap.forEach { (componentType: String, level: Int) in
```

Comment:
I think this will just get tricky since it may be possible `levelMap` will be empty, in which case the result would be optional and I've have to unwrap that. I sort of think it may be more confusing.